### ISSUE TITLE
[Bug]: UT case TestMP failed, meet DATA RACE

### ISSUE THREAD

**Issue by xzxiong** [matrixorigin/matrixone#7867]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): 
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior
pr #7860 find DATA RACE

https://github.com/matrixorigin/matrixone/actions/runs/4101924789/jobs/7074264135
![image](https://user-images.githubusercontent.com/3927687/216952331-38be4e3a-7040-474d-acbe-9430e6ba56b9.png)

```
2023-02-06T08:58:17.6389402Z === RUN   TestMP
2023-02-06T08:58:17.6389539Z ==================
2023-02-06T08:58:17.6389702Z WARNING: DATA RACE
2023-02-06T08:58:17.6389910Z Write at 0x00c000dcd5b8 by goroutine 125:
2023-02-06T08:58:17.6390262Z   github.com/matrixorigin/matrixone/pkg/common/mpool.(*fixedPool).alloc()
2023-02-06T08:58:17.6390749Z       /runner/_work/matrixone/matrixone/pkg/common/mpool/mpool.go:494 +0x94
2023-02-06T08:58:17.6391084Z   github.com/matrixorigin/matrixone/pkg/common/mpool.(*MPool).Alloc()
2023-02-06T08:58:17.6391573Z       /runner/_work/matrixone/matrixone/pkg/common/mpool/mpool.go:517 +0x168
2023-02-06T08:58:17.6391919Z   github.com/matrixorigin/matrixone/pkg/common/mpool.TestMP.func1()
2023-02-06T08:58:17.6392425Z       /runner/_work/matrixone/matrixone/pkg/common/mpool/mpool_test.go:175 +0xb0
2023-02-06T08:58:17.6392442Z
2023-02-06T08:58:17.6392669Z Previous read at 0x00c000dcd5b8 by goroutine 229:
2023-02-06T08:58:17.6393005Z   github.com/matrixorigin/matrixone/pkg/common/mpool.(*fixedPool).free()
2023-02-06T08:58:17.6393490Z       /runner/_work/matrixone/matrixone/pkg/common/mpool/mpool.go:557 +0x78
2023-02-06T08:58:17.6393821Z   github.com/matrixorigin/matrixone/pkg/common/mpool.(*MPool).Free()
2023-02-06T08:58:17.6394311Z       /runner/_work/matrixone/matrixone/pkg/common/mpool/mpool.go:583 +0x148
2023-02-06T08:58:17.6394651Z   github.com/matrixorigin/matrixone/pkg/common/mpool.TestMP.func1()
2023-02-06T08:58:17.6395153Z       /runner/_work/matrixone/matrixone/pkg/common/mpool/mpool_test.go:179 +0x90
2023-02-06T08:58:17.6395165Z
2023-02-06T08:58:17.6395360Z Goroutine 125 (running) created at:
2023-02-06T08:58:17.6395677Z   github.com/matrixorigin/matrixone/pkg/common/mpool.TestMP()
2023-02-06T08:58:17.6396177Z       /runner/_work/matrixone/matrixone/pkg/common/mpool/mpool_test.go:184 +0x148
2023-02-06T08:58:17.6396353Z   testing.tRunner()
2023-02-06T08:58:17.6396827Z       /opt/hostedtoolcache/go/1.19.5/arm64/src/testing/testing.go:1446 +0x188
2023-02-06T08:58:17.6397013Z   testing.(*T).Run.func1()
2023-02-06T08:58:17.6397488Z       /opt/hostedtoolcache/go/1.19.5/arm64/src/testing/testing.go:1493 +0x40
2023-02-06T08:58:17.6397500Z
2023-02-06T08:58:17.6397790Z Goroutine 229 (running) created at:
2023-02-06T08:58:17.6398116Z   github.com/matrixorigin/matrixone/pkg/common/mpool.TestMP()
2023-02-06T08:58:17.6398735Z       /runner/_work/matrixone/matrixone/pkg/common/mpool/mpool_test.go:184 +0x148
2023-02-06T08:58:17.6398922Z   testing.tRunner()
2023-02-06T08:58:17.6399397Z       /opt/hostedtoolcache/go/1.19.5/arm64/src/testing/testing.go:1446 +0x188
2023-02-06T08:58:17.6399595Z   testing.(*T).Run.func1()
2023-02-06T08:58:17.6400072Z       /opt/hostedtoolcache/go/1.19.5/arm64/src/testing/testing.go:1493 +0x40
2023-02-06T08:58:17.6400214Z ==================
2023-02-06T08:58:17.6400592Z panic: internal error: free size -1, possible double free
2023-02-06T08:58:17.6400608Z
2023-02-06T08:58:17.6400773Z goroutine 232 [running]:
2023-02-06T08:58:17.6401110Z github.com/matrixorigin/matrixone/pkg/common/mpool.(*fixedPool).free(0xc0002123d8, 0xc000dcd5b0)
2023-02-06T08:58:17.6401439Z    /runner/_work/matrixone/matrixone/pkg/common/mpool/mpool.go:559 +0xf8
2023-02-06T08:58:17.6401790Z github.com/matrixorigin/matrixone/pkg/common/mpool.(*MPool).Free(0xc000212380, {0xc000dcd5c0, 0x0?, 0x40})
2023-02-06T08:58:17.6402111Z    /runner/_work/matrixone/matrixone/pkg/common/mpool/mpool.go:583 +0x14c
2023-02-06T08:58:17.6402396Z github.com/matrixorigin/matrixone/pkg/common/mpool.TestMP.func1()
2023-02-06T08:58:17.6402725Z    /runner/_work/matrixone/matrixone/pkg/common/mpool/mpool_test.go:179 +0x94
2023-02-06T08:58:17.6403168Z created by github.com/matrixorigin/matrixone/pkg/common/mpool.TestMP
2023-02-06T08:58:17.6403502Z    /runner/_work/matrixone/matrixone/pkg/common/mpool/mpool_test.go:184 +0x14c
2023-02-06T08:58:17.6403780Z FAIL   github.com/matrixorigin/matrixone/pkg/common/mpool  8.071s
```

**Comment by nnsgmsone** [comment]
fixed by #7941

**Comment by xzxiong** [comment]
wait pr merged

**Comment by nnsgmsone** [comment]
mpool will be refactored next and will no longer use the difficult to maintain structure of ringbuffer. But this issue is actually very interesting, so you can learn about it if you are interested.

**Comment by nnsgmsone** [comment]
mpool will be refactored next and will no longer use the difficult to maintain structure of ringbuffer. But this issue is actually very interesting, so you can learn about it if you are interested.

**Comment by nnsgmsone** [comment]
fixed by #7867

**Comment by xzxiong** [comment]
> fixed by #7867

you means https://github.com/matrixorigin/matrixone/pull/7981, yes ?
